pkg/apis/guard/v1alpha1/queryString.go

```diff
@@ -65,10 +65,7 @@ func (pile *QueryPile) Merge(otherPile *QueryPile) {
 }
 
 func (pile *QueryPile) Clear() {
-	pile.Kv = new(KeyValPile)
-	if pile.Kv != nil {
-		pile.Kv.Clear()
-	}
+	pile.Kv = nil
```

Comment:
Align QueryString to the behavior in httpHeaders - both holders of a KeyValue Object..